python/tests/proton_tests/main.py

```diff
@@ -330,382 +356,401 @@ def filter(self, record):
 
-  def run(self):
-    return run_test(self.name(), self._run, config)
+    def run(self):
+        return run_test(self.name(), self._run, config)
 
-  def _run(self):
-    runner = Runner()
-    inst = self.cls(self.method)
-    test = getattr(inst, self.method)
+    def _run(self):
+        runner = Runner()
+        inst = self.cls(self.method)
+        test = getattr(inst, self.method)
 
-    if hasattr(inst, "configure"):
-      runner.run("configure", lambda: inst.configure(config))
-      if runner.halt(): return runner
-    if hasattr(inst, "setUp"):
-      runner.run("setup", inst.setUp)
-      if runner.halt(): return runner
+        if hasattr(inst, "configure"):
+            runner.run("configure", lambda: inst.configure(config))
+            if runner.halt():
+                return runner
+        if hasattr(inst, "setUp"):
+            runner.run("setup", inst.setUp)
+            if runner.halt():
+                return runner
 
-    runner.run("test", test)
+        runner.run("test", test)
 
-    if hasattr(inst, "tearDown"):
-      runner.run("teardown", inst.tearDown)
+        if hasattr(inst, "tearDown"):
+            runner.run("teardown", inst.tearDown)
 
-    return runner
+        return runner
+
+    def __repr__(self):
+        return "MethodTest(%r, %r)" % (self.cls, self.method)
 
-  def __repr__(self):
-    return "MethodTest(%r, %r)" % (self.cls, self.method)
 
 class PatternMatcher:
 
-  def __init__(self, *patterns):
-    self.patterns = patterns
+    def __init__(self, *patterns):
+        self.patterns = patterns
+
+    def matches(self, name):
+        for p in self.patterns:
+            if match(name, p):
+                return True
+        return False
 
-  def matches(self, name):
-    for p in self.patterns:
-      if match(name, p):
-        return True
-    return False
 
 class FunctionScanner(PatternMatcher):
 
-  def inspect(self, obj):
-    return type(obj) == types.FunctionType and self.matches(obj.__name__)
+    def inspect(self, obj):
+        return type(obj) == types.FunctionType and self.matches(obj.__name__)
+
+    def descend(self, func):
+        # the None is required for older versions of python
+        return
+        yield None
 
-  def descend(self, func):
-    # the None is required for older versions of python
-    return; yield None
+    def extract(self, func):
+        yield FunctionTest(func)
 
-  def extract(self, func):
-    yield FunctionTest(func)
 
 class ClassScanner(PatternMatcher):
 
-  def inspect(self, obj):
-    return type(obj) in CLASS_TYPES and self.matches(obj.__name__)
+    def inspect(self, obj):
+        return type(obj) in CLASS_TYPES and self.matches(obj.__name__)
 
-  def descend(self, cls):
-    # the None is required for older versions of python
-    return; yield None
+    def descend(self, cls):
+        # the None is required for older versions of python
+        return
+        yield None
```

Comment:
second occurrence of this

Reply:
Huh, isn't this the same code again?

Reply:
yes, there are three copies of this in total